chaco/grid.py

```diff
@@ -333,31 +333,59 @@ def _draw_overlay(self, gc, view_bounds=None, mode="normal"):
 
         Overrides PlotComponent.
         """
-        self._draw_component(gc, view_bounds, mode)
+        # What we're really trying to do with a grid is plot contour lines in
+        # the space of the plot.  In a rectangular plot, these will always be
+        # straight lines.
+        if not self.visible:
+            return
 
-    def overlay(self, other_component, gc, view_bounds=None, mode="normal"):
-        """Draws this component overlaid on another component.
+        if not self._cache_valid:
+            self._compute_ticks()
 
-        Overrides AbstractOverlay.
-        """
-        if not self.visible:
+        if len(self._tick_positions) == 0:
             return
-        self._compute_ticks(other_component)
-        self._draw_component(gc, view_bounds, mode)
-        self._cache_valid = False
 
-    def _draw_component(self, gc, view_bounds=None, mode="normal"):
-        """Draws the component.
+        with gc:
+            gc.set_line_width(self.line_weight)
+            gc.set_line_dash(self.line_style_)
+            gc.set_stroke_color(self.line_color_)
+            gc.set_antialias(False)
 
-        This method is preserved for backwards compatibility. Overrides
-        PlotComponent.
+            if self.component is not None:
+                gc.clip_to_rect(
+                    *(self.component.position + self.component.bounds)
+                )
+            else:
+                gc.clip_to_rect(*(self.position + self.bounds))
+
+            gc.begin_path()
+            if self.orientation == "horizontal":
+                starts = self._tick_positions.copy()
+                starts[:, 0] = self._tick_extents[:, 0]
+                ends = self._tick_positions.copy()
+                ends[:, 0] = self._tick_extents[:, 1]
+            else:
+                starts = self._tick_positions.copy()
+                starts[:, 1] = self._tick_extents[:, 0]
+                ends = self._tick_positions.copy()
+                ends[:, 1] = self._tick_extents[:, 1]
+            if self.flip_axis:
+                starts, ends = ends, starts
+            gc.line_set(starts, ends)
+            gc.stroke_path()
+
+    def overlay(self, other_component, gc, view_bounds=None, mode="normal"):
+        """Draws this component overlaid on another component.
+
+        Overrides AbstractOverlay.
         """
         # What we're really trying to do with a grid is plot contour lines in
         # the space of the plot.  In a rectangular plot, these will always be
         # straight lines.
         if not self.visible:
             return
-
+        self._compute_ticks(other_component)
```

Comment:
This method now only differs from `_draw_overlay` in this line and the `self._cache_valid = False` line as this is what `overlay` had done differently previously as well